### ISSUE TITLE
is `conda install` used for installing dependencies if a Conda environment is activated?

### ISSUE THREAD

**Issue by etijskens** [python-poetry/poetry#1776]
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have searched the [documentation](https://python-poetry.org/docs/) and believe that my question is not covered.

## Issue

As of 1.0.0 Conda environments are supported. When I install the dependencies of a project with `poetry install`, and say numpy is in the dependencies, will it use the numpy I get with `conda install numpy` or the numpy I get with `pip install numpy`?

**Comment by finswimmer** [comment]
Hello @etijskens,

`poetry` always uses `pip`. The conda support is limited to detect if your are inside a conda env and if it's the base env.

fin swimmer

**Comment by etijskens** [comment]
hello @finswimmer 
that is a pity, it also doesn't comply with the "principle of least surprise"...

**Comment by finswimmer** [comment]
Why do you want `poetry` to use `conda`?

**Comment by etijskens** [comment]
I am working for a supercomputer center. Intel Python distribution is a logical choice for HPC and based on Conda. It is great that poetry works for Conda environments, but it is a pity that the user must install dependencies himself because - for performance reasons - `Conda install numpy` is different from `pip install numpy`. From the viewpoint of the user it is unexpected that poetry install is not using Conda if the user uses a Conda environment. I am well aware that making this work is complicated, because there are different channels.

**Comment by marctorrellas** [comment]
@finswimmer, I know this is closed, but conda has other advantages, see

https://www.anaconda.com/understanding-conda-and-pip/

TLDR: 
- There is never a need to have compilers available to install them. You download binaries directly, so installation is cleaner and faster
- Additionally conda packages are not limited to Python software.

that's why I think (I already thought it was like this until recently) poetry should look at conda first if the package is available (having something like channels in conda, with Pypi being the default one)

Does that make sense at all?

**Comment by gvskalyan** [comment]
@finswimmer Is there any progress/plan in supporting packages from conda channels in addition to pypi?

**Comment by marctorrellas** [comment]
> @finswimmer Is there any progress/plan in supporting packages from conda channels in addition to pypi?

see https://github.com/python-poetry/poetry/issues/1856

**Comment by zEdS15B3GCwq** [comment]
I use miniconda as a tool for creating base virtual environments on Windows. I also used to install packages mainly from conda repositories, and pip for whatever wasn't available there, however, updating mixed environments was a big headache. There are serious discrepancies between conda repositories and pypi: the same module is sometimes packaged with different names, dependencies can be different for the same module, and conda tends to lag behind making new versions available. Mixing the two complicates things greatly. If this issue is ever reopened for reconsideration, I hope pip will remain the main installation path.
